pages/MoodEntry.tsx

```diff
@@ -10,20 +10,38 @@ type EmojiButtonProps = {
 
 const EmojiButton:FC<EmojiButtonProps> = ({ emoji, setMood }) => {
 
-  const changeBackgroundOnHover = (e:any) => {
-    e.target.style.background = 'lightblue';
+  const changeBackgroundOnHover = (e: any) => {
+    if (!e.target.className.includes(styles.clickedEmoji)) e.target.style.background = 'lightblue';
```

Comment:
This is a pattern to be really careful of. Not to say that inline styles aren't sometimes the solution, but it's rare. Here's the deal.

What styles render is determined by specificity. Different ways of applying styles apply different numbers of points of specificity. The method with the most points of specificity wins. In the biz, we call this "specificity wars". Beware specificity wars. CSS bugs are some of the nastiest ones out there, especially if you're using large shared style sheets. Making changes to increase the specificity of a selector can make something else you're not aware of, in faraway places, sharing the style, blow up dramatically.

Here's a good write-up about CSS specificity. https://stackoverflow.com/a/11934505

So, how then do you handle this in React, you ask? Good question.

Generally, you put conditional logic in the component's `className` attribute or write a CSS selector for the case. For example, let's say you want to style on a checked state. You have a few ways you can do something like this.

## Hidden Checkboxes and the CSS Solution

This is a pattern that we used for years and years and years before the days of React. It's kind of slick.

```typescript
<form>
  <label>
    <input type="checkbox" className="ye-checkbox">
    <span className="ye-text-label">Toggle me!</span>
  </label>
</form>
```

```css
.ye-text-label {
  background: red;
  padding: .375rem .5rem;
}

.ye-checkbox:checked + .ye-text-label {
  background: green;
}

.ye-checkbox {
  display: none;
}
```

This uses the checkbox on the DOM as state and styles the `span` based on its checked value. It was a fairly elegant way to store and use state in CSS. This is rarely used these days, as it's super easy in React to declare some state and use it in your interfaces, like this:

## React and Turnarys or Conditional Renders in Class Names

This pattern is what you'll see most of the time these days. It goes like this. You have some state. In the render method, you set the classes that the element will have based on the state (or lack thereof). Something like this:

```typescript
type Props = {
  value: string
}

const MyComponent: FC<Props> = ({value}) => {
  const [anotherValue, setAnotherValue] = useState()

  return (
    <div className={`${value ? "class-one" : "class-two"} ${anotherValue && "class-three"}`}>{value}</div>
  )
}
```

This means that you can keep your CSS shallow and specified by `class`. Ideally, you only ever have 1 or 2 points of specificity, which means that you never have to go nuclear on your CSS.

CSS modules also help (which Next supports out of the gate), though it means that you have to implement a little differently to make the above scenario work. For CSS modules, it helps to build or pull a helper function that gracefully concatenates CSS class names (usually it's called `classNames` or something like that). Here's one that gets used all the freaking time. https://www.npmjs.com/package/classnames. I tend to write my own, as I don't feel like the addition of a dependency for this really merits the increase in exposure and potential for problems that you get by adding a dependency.

Reply:
I want to go the class route, but I'm still a bit iffy on implementation. Noted + check later.